deps/ipc-util/ipc-util/pipe-windows.h

```diff
@@ -30,13 +31,55 @@ struct ipc_pipe_server {
 
 	ipc_pipe_read_t read_callback;
 	void *param;
+
+	ipc_pipe_connect_t connect_callback;
+	void *connect_param;
+
+	ipc_pipe_disconnect_t disconnect_callback;
+	void *disconnect_param;
+
+	bool is_return;
 };
 
 struct ipc_pipe_client {
+	OVERLAPPED overlap;
 	HANDLE handle;
+	HANDLE ready_event;
+	HANDLE thread;
+
+	uint8_t *read_data;
+	size_t size;
+	size_t capacity;
+
+	ipc_pipe_read_t read_callback;
+	void *param;
+
+	ipc_pipe_disconnect_t disconnect_callback;
+	void *disconnect_param;
+
+	bool is_return;
 };
```

Comment:
I wrote it split up because I wanted to keep as much of the original code in place to not accidentally break things (like with Game Capture). Now that I'm far more confident with how things are working, I can probably do just that, yes.